### PR TITLE
Uar 168 udate edit entity demo 

### PR DESCRIPTION
**Pull request by AndreKramer7654**
Roll up branch for sprint demo

### CODE REVIEW

File: src/model/application.model.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { CreatePaymentRequest } from "@companieshouse/api-sdk-node/dist/services/payment";
+import { ICompanyDetails } from "./company.profile.model";
```
Comment (bantao-ch) [inline src/model/application.model.ts:2]: This should be CompanyProfile from the node-sdk-api !! hence the build is breaking.

File: src/model/application.model.ts [hunk 2]
```diff
@@ -34,7 +35,9 @@ export interface ApplicationData {
     has_sold_land?: string;
     is_secure_register?: string;
     who_is_registering?: string;
+    company_profile_details?: ICompanyDetails;
```
Comment (bantao-ch) [inline src/model/application.model.ts:38]: Correspondingly change this to CompanyProfile

File: test/__mocks__/session.mock.ts [hunk 3]
```diff
@@ -864,6 +868,21 @@ export const TRUST_PARTIAL_DATE: Trust = {
   unable_to_obtain_all_trust_info: "No"
 };
 
+export const OVER_SEAS_ENTITY_MOCK_DATA: ICompanyDetails = {
```
Comment (bantao-ch) [inline test/__mocks__/session.mock.ts:871]: This would need to change to CompanyProfile !

File: src/controllers/update/overseas.entity.review.controller.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  } catch (errors) {
+    logger.errorRequest(req, errors);
+    next(errors);
+  }
+};
+
+export const post = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `POST ${config.UPDATE_OVERSEAS_ENTITY_REVIEW_PAGE}`);
+
```
Comment (AndreKramer7654) [inline src/controllers/update/overseas.entity.review.controller.ts:30]: Delete commented out code

File: src/model/company.profile.model.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    // serviceAddress?: Address;
+    // email?: string;
+    // legalForm?: string;
+    // lawGoverned?: string;
+    // publicRegisterName?: string;
+    // publicRegisterJurisdiction?: string;
+    // registrationNumber?: string;
+    // isOnRegisterInCountryFormedIn?: yesNoResponse;
+}
+
```
Comment (bantao-ch) [inline src/model/company.profile.model.ts:21]: It would be good to clean up by deleting this file so that the inconsistencies can be easily resolved as we are using the CompanyProfile from node-sdk-api

File: src/controllers/update/overseas.entity.update.details.controller.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    // Wipe 'register in country formed in' data if IsOnRegisterInCountryFormedInKey is no or not selected
+    if (!data[IsOnRegisterInCountryFormedInKey]) {
+      data[PublicRegisterNameKey] = '';
+      data[PublicRegisterJurisdictionKey] = '';
+      data[RegistrationNumberKey] = '';
+    }
+
+    const session = req.session as Session;
+    setApplicationData(session, data, EntityKey);
+    // await saveAndContinue(req, session);
```
Comment (AndreKramer7654) [inline src/controllers/update/overseas.entity.update.details.controller.ts:77]: for save/resume - remove commented out code for now?

File: src/model/company.profile.model.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Address } from "model/data.types.model";
+export interface ICompanyDetails {
```
Comment (AndreKramer7654) [inline src/model/company.profile.model.ts:2]: remove this as now using CompanyProfile

File: src/routes/index.ts [hunk 8]
```diff
@@ -122,6 +129,16 @@ router.post(config.MANAGING_OFFICER_CORPORATE_URL, authentication, navigation.ha
 router.post(config.MANAGING_OFFICER_CORPORATE_URL + config.ID, authentication, navigation.hasBeneficialOwnersStatement, ...validator.managingOfficerCorporate, checkValidations, managingOfficerCorporate.update);
 router.get(config.MANAGING_OFFICER_CORPORATE_URL + config.REMOVE + config.ID, authentication, navigation.hasBeneficialOwnersStatement, managingOfficerCorporate.remove);
 
+//  FETCH AND UPDATE COMPANY PROFILE
```
Comment (AndreKramer7654) [inline src/routes/index.ts:132]: slightly misleading comment as first to follow is the confirm page. Also comment should not be all upper case.

File: src/routes/index.ts [hunk 9]
```diff
@@ -122,6 +129,16 @@ router.post(config.MANAGING_OFFICER_CORPORATE_URL, authentication, navigation.ha
...
+router.get(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL, authentication, confirmOverseasEntityDetails.get);
+router.post(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL, authentication, confirmOverseasEntityDetails.post);
+
+router.get(config.UPDATE_OVERSEAS_ENTITY_REVIEW_URL, authentication, updateOverseasEntityReview.get);
```
Comment (AndreKramer7654) [inline src/routes/index.ts:136]: This review route is a bit lost in between others. Re-order?

File: src/service/overseas.entities.service.ts [hunk 10]
```diff
@@ -58,6 +58,22 @@ export const updateOverseasEntity = async (req: Request, session: Session) => {
   logger.debugRequest(req, `Updated Overseas Entity, ${JSON.stringify(response)}`);
 };
 
+export const getCompanyRequest = async (
```
Comment (AndreKramer7654) [inline src/service/overseas.entities.service.ts:61]: Please see comments in #490 on this.

File: src/utils/update/company.profile.mapper.to.oversea.entity.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    principal_address: {
+      property_name_number: cp.registeredOfficeAddress?.premises,
+      line_1: cp.registeredOfficeAddress?.addressLineOne,
+      line_2: cp.registeredOfficeAddress?.addressLineTwo,
+      town: cp.registeredOfficeAddress?.locality,
+      county: cp.registeredOfficeAddress?.region,
+      country: cp.registeredOfficeAddress?.country,
+      postcode: cp.registeredOfficeAddress?.postalCode
+    }
+  // is_on_register_in_country_formed_in: isOnReg // breaks node
```
Comment (AndreKramer7654) [inline src/utils/update/company.profile.mapper.to.oversea.entity.ts:26]: maybe we should note some fields missing and remove 26/27 comments.

File: src/utils/update/company.profile.mapper.to.oversea.entity.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile/types";
+import { Entity } from "model/entity.model";
+
+export const mapCompanyProfileToOverseasEntityToDTOUpdate = (cp: CompanyProfile): Entity => {
+  if (!cp){
```
Comment (AndreKramer7654) [inline src/utils/update/company.profile.mapper.to.oversea.entity.ts:5]: Better to fail than to return and empty Entity. Remove if

File: src/validation/error.messages.ts [hunk 13]
```diff
@@ -107,6 +107,7 @@ export enum ErrorMessages {
   MAX_PUBLIC_REGISTER_NAME_LENGTH = "Name of register must be 160 characters or less",
   MAX_PUBLIC_REGISTER_NUMBER_LENGTH = "Registration number must be 160 characters or less",
   MAX_OCCUPATION_LENGTH = "Occupation must be 100 characters or less",
+  MAX_OE_LENGTH = "OE number must be 8 characters",
```
Comment (AndreKramer7654) [inline src/validation/error.messages.ts:110]: This is a left over from previous. Now INVALID_OE_NUMBER is tested for

File: test/__mocks__/session.mock.ts (hunk 3 above)
Comment (AndreKramer7654) [reply test/__mocks__/session.mock.ts:871]: just remove for now

File: test/__mocks__/text.mock.ts [hunk 14]
```diff
@@ -127,4 +127,9 @@ export const CHANGE_LINK_INDIVIDUAL_BO_IS_ON_SANCTIONS_LIST = CHANGE_LINK_INDIVI
 
 // Update journey
 export const OVERSEAS_ENTITY_QUERY_PAGE_TITLE = "What is the OE number?";
-export const OE_NUMBER_FIELD_POPULATED = "value=\"OE123456\"";
+export const CONFIRM_OVERSEAS_ENTITY_PAGE_TITLE = "Confirm this is the registered overeseas entity";
+export const UPDATE_OVERSEAS_ENTITY_PAGE_TITLE = "Confirm or update overseas entity details";
+export const CONFIRM_AND_CONTINUE_BUTTON_TEXT = "Confirm and continue";
+export const CHANGE_COMPANY_TEST = "Change company";
```
Comment (AndreKramer7654) [inline test/__mocks__/text.mock.ts:133]: Remove 132. Not testing company changes.

File: views/update/confirm-overseas-entity-details.html [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+          attributes: {
+            "id": "oe_entity_address_value"
+          }
+        }
+      ],
+      [
+        {
+          text: "Registered On",
+          attributes: {
+            "id": "oe_entity_registered_date"
```
Comment (AndreKramer7654) [inline views/update/confirm-overseas-entity-details.html:57]: This can be an additional thing we pass in from the session. Stored separately from the Entity.

Comment (AndreKramer7654) [reply views/update/confirm-overseas-entity-details.html:57]: We can discuss the need for this field separately

File: views/update/includes/entity.html [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+      {
+        key: {
+          text: "Name of the overseas entity"
+        },
+        value: {
+          text: appData.entity.name
+        },
+        actions: {
+          items: [ CREATE_UPDATE_CHANGE_LINK(
+            OE_CONFIGS.UPDATE_ENTITY_URL,
```
Comment (AndreKramer7654) [inline views/update/includes/entity.html:36]: This should not be a constant (OE_CONFIGS.UPDATE_ENTITY_UR) but driven from the session. Then we can re-use the page in the registration journey too. Once we prove that this view would replace the registration Entity one and we can use it for both journeys.

File: views/update/includes/entity.html [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+      {
+        key: {
+          text: "Country it was formed in"
+        },
+        value: {
+          text: appData.entity.incorporation_country
+        },
+        actions: {
+          items: [CREATE_UPDATE_CHANGE_LINK(
+            OE_CONFIGS.UPDATE_ENTITY_URL,
```
Comment (AndreKramer7654) [inline views/update/includes/entity.html:52]: same here as above and similar below

File: views/update/overseas-entity-query.html [hunk 18]
```diff
@@ -12,12 +12,26 @@
...
+                errorList: [
+                {
+                  text: errorList,
+                  href: "#oe_number"
+                }
+              ]
+            }) 
+            }}
+        {% endif %}
+                {% include "includes/list/errors.html" %}
```
Comment (AndreKramer7654) [inline views/update/overseas-entity-query.html:34]: whitespace formatting?

File: src/service/retry.handler.service.ts [hunk 19]
```diff
@@ -54,11 +54,8 @@ export const makeApiCallWithRetry = async (
 
     client = createOAuthApiClient(session);
     response = await client[serviceName][fnName](...otherParams);
-
   }
-
   logger.debugRequest(req, 'Call successful.');
-
   return response;
```
Comment (corybudd-ch) [inline src/service/retry.handler.service.ts:59]: any need for these line changes?

File: src/utils/application.data.ts [hunk 20]
```diff
@@ -97,3 +97,6 @@ const getIndexInApplicationData = (req: Request, appData: ApplicationData, key:
     throw createAndLogErrorRequest(req, `application.data getIndexInApplicationData - unable to find object in session data for key ${key} and ID ${id}`);
   }
 };
+
+
+
```
Comment (corybudd-ch) [inline src/utils/application.data.ts:102]: any need for these line changes?

File: src/routes/index.ts [hunk 21]
```diff
@@ -30,9 +30,12 @@ import {
...
   accessibilityStatement,
   signOut,
-  trustDetails,
+  resumeSubmission,
   trustInvolved,
-  resumeSubmission
+  confirmOverseasEntityDetails,
+  updateOverseasEntityReview,
+  updateOverseasEntityDetails,
+  trustDetails
```
Comment (corybudd-ch) [inline src/routes/index.ts:38]: unnecessary line changes here too with trustDetails and resumeSubmission moving

File: test/__mocks__/session.mock.ts [hunk 22]
```diff
@@ -18,12 +18,15 @@ import {
   managingOfficerType,
   presenterType,
   trustType,
-  dueDiligenceType
+  dueDiligenceType,
```
Comment (corybudd-ch) [inline test/__mocks__/session.mock.ts:21]: unnecessary change here too

File: test/controllers/update/overseas.entity.query.controller.spec.ts [hunk 23]
```diff
@@ -59,7 +58,7 @@ describe("OVERSEAS ENTITY QUERY controller", () => {
       const resp = await request(app).get(config.OVERSEAS_ENTITY_QUERY_URL);
 
       expect(resp.status).toEqual(200);
-      expect(resp.text).toContain(OE_NUMBER_FIELD_POPULATED);
+      // expect(resp.text).toContain(OE_NUMBER_FIELD_POPULATED);
```
Comment (corybudd-ch) [inline test/controllers/update/overseas.entity.query.controller.spec.ts:61]: why has this assertion been removed?

File: test/__mocks__/session.mock.ts (hunk 22 above)
Comment (olaych) [reply test/__mocks__/session.mock.ts:21]: Resolved but note this is a branch that has 3 large branches in it, difficult to track changes

File: views/update/includes/entity.html (hunk 17 above)
Comment (olaych) [reply views/update/includes/entity.html:52]: I think this should be fixed in the local branches, same as above

File: src/utils/update/company.profile.mapper.to.oversea.entity.ts (hunk 12 above)
Comment (olaych) [reply src/utils/update/company.profile.mapper.to.oversea.entity.ts:5]: But shouldn't we be handling the fail? An undefined data will generate a 500 error and need to demonstrate unit test in place to handle it. Checking through existing CH mapping, similar approach is in place

File: src/model/application.model.ts (hunk 2 above)
Comment (olaych) [reply src/model/application.model.ts:38]: 👍🏿

File: test/__mocks__/session.mock.ts (hunk 22 above)
Comment (corybudd-ch) [reply test/__mocks__/session.mock.ts:21]: No worries, I  agree ❤️

File: src/utils/update/company.profile.mapper.to.oversea.entity.ts (hunk 12 above)
Comment (AndreKramer7654) [reply src/utils/update/company.profile.mapper.to.oversea.entity.ts:5]: Why are you expecting it to fail (be undefined or null)? We should always get back a CompanyProfile as the SDK should always return a result.

File: src/controllers/update/company.profile.controller.ts [hunk 24]
```diff
@@ -14,10 +14,9 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
     logger.debugRequest(req, `GET ${config.CONFIRM_OVERSEAS_ENTITY_DETAILS_PAGE}`);
     const session = req.session as Session;
     const appData: ApplicationData = getApplicationData(session);
-    const id: string = appData?.oe_number || "";
-    const companyDataResponse = await getCompanyRequest(req, id);
-    const overseasEntity = mapCompanyProfileToOverseasEntityToDTOUpdate(companyDataResponse as CompanyProfile);
-
+    const id: string | any = appData?.oe_number;
+    const companyDataResponse = await getCompanyRequest(req, id) as CompanyProfile;
+    const overseasEntity = mapCompanyProfileToOverseasEntityToDTOUpdate(companyDataResponse);
```
Comment (AndreKramer7654) [inline src/controllers/update/company.profile.controller.ts:19]: move to after if statement. Line 22

File: src/utils/update/company.profile.mapper.to.oversea.entity.ts [hunk 25]
```diff
@@ -1,62 +1,30 @@
...
-//     // serviceAddress: mapCompanyAddressToOverseasEntityAddressDTO(data?.registeredOfficeAddress),
-//     // legalForm: data?.legal_form,
-//     // lawGoverned: data?.foreign_company_details?.governed_by,
-//     // publicRegisterJurisdiction: data?.jurisdiction,
-//     // registrationNumber: data?.registration_number,
-//     // isServiceAddressSameAsPrincipalAddress: mapYesOrNo(data)
-//   };
-// };
-
 export const mapCompanyProfileToOverseasEntityToDTOUpdate = (cp: CompanyProfile): Entity => {
```
Comment (AndreKramer7654) [inline src/utils/update/company.profile.mapper.to.oversea.entity.ts:4]: rename  mapCompanyProfileToOverseasEntity() it's not a DTO. That part is done in the api SDK.

**Comment by AndreKramer7654** [comment]
Only for sprint demo prep
